### PR TITLE
 New Cask Raven Lite 2.0 sound analysis software

### PR DESCRIPTION
**Pull request by BenjaminHCCarr**
This adds the 64 bit Free version of the Cornell Lab of Ornithology Raven Lite 2.0 software used "for the acquisition, visualization, measurement, and analysis of sounds."
After making all changes to the cask:

- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` reports no offenses.
- [X] The commit message includes the cask’s name and version.

Additionally, **if adding a new cask**:

- [X] Named the cask according to the [token reference].
- [X] `brew cask install {{cask_file}}` worked successfully.
- [X] `brew cask uninstall {{cask_file}}` worked successfully.
- [X] Checked there are no [open pull requests] for the same cask.
- [X] Checked that the cask was not already refused in [closed issues].

### CODE REVIEW

File: Casks/raven-lite.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+cask 'raven-lite' do
+  version '2.0.0.0022'
+  sha256 '84c8d97df6119605ff645c93d896d472a814246fb87128611225168459423532'
+
+  url "http://www.birds.cornell.edu/brp/RavenLite/RavenLiteFullInstaller/InstData/MacOSX/RavenLite-#{version}_macosx.dmg"
+  name 'Raven Lite 2.0'
```
Comment (miccal) [inline Casks/raven-lite.rb]: Please change this to `name "Raven Lite #{version.major-minor}"`.

File: Casks/raven-lite.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  homepage 'http://www.birds.cornell.edu/brp/RavenLite/RavenLiteReadMe.htm'
```
Comment (miccal) [inline Casks/raven-lite.rb]: Please change this to `homepage 'http://www.birds.cornell.edu/brp/raven/RavenOverview.html'`.

File: Casks/raven-lite.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+  depends_on macos: '>= :mountain_lion'
+  depends_on arch: :x86_64
+
+  installer manual: 'RavenLite-Installer.app'
+
+  uninstall trash: '/Applications/Raven Lite 2.0/'
```
Comment (miccal) [inline Casks/raven-lite.rb]: Please change this to `uninstall delete: "/Applications/Raven Lite #{version.major-minor}"`.

File: Casks/raven-lite.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+  caveats do
+    depends_on_java('6')
```
Comment (miccal) [inline Casks/raven-lite.rb]: The website `http://www.birds.cornell.edu/brp/RavenLite/RavenLiteReadMe.htm` states:

> Raven Lite requires a Java runtime environment version 8 update 40 or newer.

Are you sure Java 6 is sufficient?

File: Casks/raven-lite.rb (hunk 1 above)
Comment (vitorgalvao) [inline Casks/raven-lite.rb]: Small correction here. Name should actually be `name 'Rave Lite'`. Versions are not part of names.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/raven-lite.rb (hunk 4 above)
Comment (vitorgalvao) [inline Casks/raven-lite.rb]: > Are you sure Java 6 is sufficient?

It’s not even a matter of it being *sufficient*, as it is now, it says that version is a *requirement*. For it to be minimum, it should just be `depends_on_java`. [Docs](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/caveats.md#caveats-mini-dsl):

| method                            | description |
| --------------------------------- | ----------- |
| `depends_on_java(version)`        | users should make sure they have the specified version of java installed. `version` can be exact (e.g. `6`), a minimum (e.g. `7+`), or omitted (when any version works).

File: Casks/raven-lite.rb (hunk 3 above)
Comment (vitorgalvao) [inline Casks/raven-lite.rb]: It’s highly unlikely moving this app into place is the only thing the installer does. It needs either more things to uninstall or an official page stating this really is the only modification.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/raven-lite.rb (hunk 1 above)
Comment (miccal) [inline Casks/raven-lite.rb]: Thanks @vitorgalvao, I did not realise that.

File: Casks/raven-lite.rb (hunk 4 above)
Comment (miccal) [inline Casks/raven-lite.rb]: Thanks @vitorgalvao - by _sufficient_ I meant should it be a higher version.

File: Casks/raven-lite.rb (hunk 1 above)
Comment (BenjaminHCCarr) [inline Casks/raven-lite.rb]: Thanks for comments, the reason I included the 2.0 is that 1.0 still exists for i386 and other systems, it was referred to in the community as Raven Lite 2 for many years, maybe that has gone by the wayside.

File: Casks/raven-lite.rb (hunk 2 above)
Comment (BenjaminHCCarr) [inline Casks/raven-lite.rb]: This is problematic, as Raven is actually 3 programs; lite [free], pro [licensed for "real" money], and exhibit; the demonstration or interactive museum type system.

I have the other two casked, put this one in last night to work out any kinks, and see that you have both found some, so glad we're going to get this all worked out in one thread.

File: Casks/raven-lite.rb (hunk 4 above)
Comment (BenjaminHCCarr) [inline Casks/raven-lite.rb]: @vitorgalvao the stanzas section is out of date then:
[cask_language_reference/stanzas/depends_on.md](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/depends_on.md)
as it says java is not implemented yet. I have always been working out of the cask stanzas docs.

Sorry about the `6` will fix to `8`.

I believe I made the poor assumption that all three would depend on the same min version.

Comment (BenjaminHCCarr) [inline Casks/raven-lite.rb]: I see you linked to the caveats section, I see, will update to `8+`

Comment (vitorgalvao) [inline Casks/raven-lite.rb]: @BenjaminHCCarr As you seem to have gathered already, `depends_on_java` is not the same as `depends_on java:`.

File: Casks/raven-lite.rb (hunk 3 above)
Comment (BenjaminHCCarr) [inline Casks/raven-lite.rb]: @miccal the stanzas state that trash is currently stubbed to delete; isn't having trash going to be the move going foward or is @vitorgalvao going to script all of them in one go at some point (eg: sha256).

There is an uninstall script, it is a a jar, and I was unable to get `uninstall script:` or `uninstall executable:` to run it.
The jar simply iterates through a file called `install.log` here is the first 5 and last 5 of that file:

```
 head -n 5 install.log && tail -n 5 install.log 
/Applications/Raven Lite 2.0
/Applications/Raven Lite 2.0/readme_lite.htm
/Applications/Raven Lite 2.0/DefaultRavenPreferences.txt
/Applications/Raven Lite 2.0/NewFeatureSites.txt
/Applications/Raven Lite 2.0/RavenPreferencesMasterList.txt
/Applications/Raven Lite 2.0/Examples/RBnuthatch.wav
/Applications/Raven Lite 2.0/Examples/Raven.aif
/Applications/Raven Lite 2.0/Examples/SpottedHyena.aif
/Applications/Raven Lite 2.0/.installationinformation
/Applications/Raven Lite 2.0/Uninstaller/uninstaller.jar
```

The last line being nuking the uninstall jar.

`/Applications/Raven Lite 2.0/Uninstaller/uninstaller.jar` is the uninstaller script I unzipped and all it does is remove the directory.

I even tried variations on `'open /Applications/Raven Lite 2.0/Uninstaller/uninstaller.jar'` with `script:` and `executable:` with no luck.

none of the pkg recommendations apply as it is install with a `.app` with none of the pkg recores I could find.

I am happy to do it the "correct" way, I just was unable to find any way to run a java jar as an uninstall; and as `delete`|`trash` served the same function that the jar performed I used that as the uninstall to compliment `install manual:`

**Comment by BenjaminHCCarr** [comment]
Per comments from @miccal and @vitorgalvao

1) I update the application name to not include a version
2) changed the download version to be #{version.major_minor_patch} (which had failed for me initially but that was a ruby single/double quote thing)
3) update java caveat from my "stub" cask formula for another raven product to reflect `8+`

- did not change `trash` to `delete`; awaiting further comment; is there a way to run a jar as an uninstall?

None of these worked:
* `uninstall script: '/Applications/Raven Lite 2.0/Uninstaller/uninstaller.jar'`
* `uninstall script: 'open /Applications/Raven Lite 2.0/Uninstaller/uninstaller.jar'`
* `uninstall executable: '/Applications/Raven Lite 2.0/Uninstaller/uninstaller.jar'`
* `uninstall executable: 'open /Applications/Raven Lite 2.0/Uninstaller/uninstaller.jar'`

so based on [cask_language_reference/stanzas/uninstall.md](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/uninstall.md)

I went with `trash` over `delete` as it is a synoym, and based on the "future" comment seemed to be preemptive in the way casks were moving.

File: Casks/raven-lite.rb (hunk 2 above)
Comment (miccal) [inline Casks/raven-lite.rb]: Yes but there should be an obvious link to the download from the `homepage`, and your current one has no links to actually download the `.dmg`.

File: Casks/raven-lite.rb (hunk 3 above)
Comment (miccal) [inline Casks/raven-lite.rb]: So the uninstall script simply deletes every file in `/Applications/Raven Lite 2.0`?

File: Casks/raven-lite.rb (hunk 4 above)
Comment (miccal) [inline Casks/raven-lite.rb]: Yes, `depends_on_java` is a [`caveat`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/caveats.md) not a [`depends_on`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/depends_on.md).

File: Casks/raven-lite.rb (hunk 3 above)
Comment (BenjaminHCCarr) [inline Casks/raven-lite.rb]: @miccal yes, it's just a jar file that removes the install directory `Applications/Raven Lite 2.0` based on the input of `install.log`.

I have a feeling this is just for packaging the `.jar` for cross-platform use, the script input from `install.log` would be for `C:\Program files\Raven Lite 2.0` on Windows (or something like that)  and likely `/opt/local/Raven Lite 2.0/` on linux.

Comment (miccal) [inline Casks/raven-lite.rb]: In that case then I would say removing `/Applications/Raven Lite 2.0/` is enough.

File: Casks/raven-lite.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+cask 'raven-lite' do
+  version '2.0.0.0022'
+  sha256 '84c8d97df6119605ff645c93d896d472a814246fb87128611225168459423532'
+
+  url "http://www.birds.cornell.edu/brp/RavenLite/RavenLiteFullInstaller/InstData/MacOSX/RavenLite-#{version}_macosx.dmg"
+  name 'Raven Lite'
+  homepage 'http://www.birds.cornell.edu/brp/RavenLite/RavenLiteReadMe.htm'
```
Comment (miccal) [inline Casks/raven-lite.rb]: Please change this to `homepage 'http://www.birds.cornell.edu/brp/raven/RavenOverview.html'`. I know you stated why you thought this `homepage` was more suitable, but I think there should be an obvious link to the download from the `homepage`, and your current one has no links to actually download the `.dmg`.

File: Casks/raven-lite.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+  depends_on macos: '>= :mountain_lion'
+  depends_on arch: :x86_64
+
+  installer manual: 'RavenLite-Installer.app'
+
+  uninstall trash: '/Applications/Raven Lite 2.0/'
+
+  caveats do
+      depends_on_java('8+')
```
Comment (miccal) [inline Casks/raven-lite.rb]: Please remove two spaces.

**Comment by BenjaminHCCarr** [comment]
1. I was able to find the launcher buried in the app installer, I have updated that here, that means we no longer get the you 'you must run .app' output which was really annoying me
2. Updated homepage to the main Raven page
3. changed uninstall to delete

* this codeblock would make the uninstaller script possible


    uninstall system_command: "\`which java\` -jar /Applications/Raven Lite 2.0/Uninstaller/uninstaller.jar",
    sudo: false

also oddly, in the `uninstaller` command, passing `sudo: false` warns with

> ==> Running uninstall process for raven-lite; your password may be necessary
> Warning: Unknown arguments to uninstall -- [:sudo]. Running "brew update; brew cleanup; brew cask cleanup" will likely fix it.

File: Casks/raven-lite.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  sha256 '84c8d97df6119605ff645c93d896d472a814246fb87128611225168459423532'
+
+  url "http://www.birds.cornell.edu/brp/RavenLite/RavenLiteFullInstaller/InstData/MacOSX/RavenLite-#{version}_macosx.dmg"
+  name 'Raven Lite'
+  homepage 'http://www.birds.cornell.edu/brp/raven/RavenOverview.html'
+
+  depends_on macos: '>= :mountain_lion'
+  depends_on arch: :x86_64
+
+  installer script: './RavenLite-Installer.app/Contents/MacOS/JavaAppLauncher', sudo: false
```
Comment (miccal) [inline Casks/raven-lite.rb]: This should be:
```
installer script: 'RavenLite-Installer.app/Contents/MacOS/JavaAppLauncher',
          sudo:   false
```

File: Casks/raven-lite.rb [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
```
Comment (miccal) [inline Casks/raven-lite.rb]: Please remove the extra whitespace from this line.

**Comment by miccal** [comment]
Thank you @BenjaminHCCarr for accomodating all the requested changes.

**Review by miccal — APPROVED** [review]

File: Casks/raven-lite.rb [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+  version '2.0.0.0022'
+  sha256 '84c8d97df6119605ff645c93d896d472a814246fb87128611225168459423532'
+
+  url "http://www.birds.cornell.edu/brp/RavenLite/RavenLiteFullInstaller/InstData/MacOSX/RavenLite-#{version}_macosx.dmg"
+  name 'Raven Lite'
+  homepage 'http://www.birds.cornell.edu/brp/raven/RavenOverview.html'
+
+  depends_on macos: '>= :mountain_lion'
+
+  installer script: 'RavenLite-Installer.app/Contents/MacOS/JavaAppLauncher', sudo: false
```
Comment (vitorgalvao) [inline Casks/raven-lite.rb]: Won’t this call a GUI?

**Review by miccal — APPROVED** [review]

**Pull request merged**